src/coreclr/jit/optimizer.cpp

```diff
@@ -5166,40 +5193,110 @@ bool Compiler::optIsLoopClonable(unsigned loopInd)
 #endif // JIT32_GCENCODER
     if (fgReturnCount + loopRetCount > epilogLimit)
     {
-        JITDUMP("Loop cloning: rejecting loop because it has %d returns; if added to previously-existing %d returns, "
-                "would exceed the limit of %d.\n",
-                loopRetCount, fgReturnCount, epilogLimit);
+        JITDUMP("Loop cloning: rejecting loop L%02u. It has %d returns;"
+                " if added to previously existing %d returns, it would exceed the limit of %d.\n",
+                loopInd, loopRetCount, fgReturnCount, epilogLimit);
+        return false;
```

Comment:
Don't forget https://github.com/dotnet/runtime/issues/48740#issuecomment-785938375

Reply:
fyi, I had previously added this to the list of loop work items: https://github.com/dotnet/runtime/issues/43549